From 9564c22a92b703fd334fa7c49fb40119ca983ef5 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 15 Feb 2024 08:25:56 +0200
Subject: [PATCH] [FSW-402] fs1: add device specific memory defines

This patch adds device specific memory region defines
in header file.

Change-Id: I882cec020540851365f197ecc4345dcdb012bb46
---

diff --git a/arch/fs1/arc/soc.h b/arch/fs1/arc/soc.h
new file mode 100644
index 0000000..352ba9d
--- /dev/null
+++ b/arch/fs1/arc/soc.h
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2024 HabanaLabs Ltd.
+ */
+
+#ifndef _SOC_H_
+#define _SOC_H_
+
+enum region_idx {
+	REGION_ICCM_IDX = 0,
+	REGION_HBW_0_IDX = 1,
+	REGION_HBW_1_IDX = 2,
+	REGION_HBW_2_IDX = 3,
+	REGION_HBW_3_IDX = 4,
+	REGION_HBW_4_IDX = 5,
+	REGION_HBW_5_IDX = 6,
+	REGION_FW_MEM_IDX = 7,
+	REGION_DCCM_IDX = 8,
+	REGION_GEN_MEM_IDX = 9,
+	REGION_GT_TILE_0_IDX = 10,
+	REGION_GT_TILE_1_IDX = 11,
+	REGION_GT_TILE_2_IDX = 12,
+	REGION_GT_TILE_3_IDX = 13,
+	REGION_LBW_DBG_IDX = 14,
+	REGION_LBW_CFG_IDX = 15,
+};
+
+#define REGION_BASE_ADDR(REG_IDX)	(REG_IDX << 28)
+#define SZ_32K				0x00008000
+#define SZ_64K				0x00010000
+#define SZ_128K				0x00020000
+#define SZ_256K				0x00040000
+#define SZ_4MB				0x00400000
+#define SZ_256MB			0x10000000
+
+#define FLASH_VALID_MASK		0x1FFFFF
+
+#define HL_CFG_BASE			REGION_BASE_ADDR(REGION_LBW_CFG_IDX)
+#define HL_CFG_SIZE			SZ_256MB   /* 256MB */
+
+#define HL_ICCM_BASE			REGION_BASE_ADDR(REGION_ICCM_IDX)
+#define HL_ICCM_SIZE			0x40000   /* 256KB */
+
+#define HL_DCCM_BASE			REGION_BASE_ADDR(REGION_DCCM_IDX)
+#define HL_DCCM_SIZE			0x20000   /* 128KB */
+
+#define HL_FW_MEM_REGION_BASE		REGION_BASE_ADDR(REGION_FW_MEM_IDX)
+#define HL_FW_MEM_OFFSET		0x8800000
+#define HL_FW_MEM_BASE			(HL_FW_MEM_REGION_BASE + HL_FW_MEM_OFFSET)
+#define HL_FW_MEM_SIZE			SZ_4MB
+
+#define HL_GEN_MEM_BASE			REGION_BASE_ADDR(REGION_GEN_MEM_IDX)
+#define HL_FLASH_OFFSET			0x4000000
+
+#define HL_ARC0_ICCM_OFFSET		0x0000000
+#define HL_ARC0_DCCM_OFFSET		0x0010000
+#define HL_ARC1_ICCM_OFFSET		0x0040000
+#define HL_ARC1_DCCM_OFFSET		0x0080000
+#define HL_ARC2_ICCM_OFFSET		0x00C0000
+#define HL_ARC2_DCCM_OFFSET		0x00E0000
+#define HL_ICCM_OFFSET			HL_ARC1_ICCM_OFFSET
+#define HL_SBROM_OFFSET			0x0180000
+#define HL_SCRATCHPAD_SRAM_OFFSET	0x0200000
+#define HL_FWSRAM_OFFSET		0x01A0000
+#define HL_PCIE_FWSRAM_OFFSET		0x0380000
+#define SP_SRAM_BASE			(HL_FW_MEM_BASE + HL_SCRATCHPAD_SRAM_OFFSET)
+
+#define HL_FW_MEM_BASE_FULL		0x300007FF8800000ull
+#define HL_SP_SRAM_BASE			(HL_FW_MEM_BASE_FULL + HL_SCRATCHPAD_SRAM_OFFSET)
+#define HL_SP_SRAM_SIZE			0x100000
+
+#define HL_ARC0_ICCM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_ARC0_ICCM_OFFSET)
+#define ARC0_ICCM_SIZE			SZ_64K
+#define ARC0_DCCM_SIZE			SZ_32K
+#define HL_ARC1_ICCM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_ARC1_ICCM_OFFSET)
+#define ARC1_ICCM_SIZE			SZ_256K
+#define ARC1_DCCM_SIZE			SZ_128K
+#define HL_ARC2_ICCM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_ARC2_ICCM_OFFSET)
+#define ARC2_ICCM_SIZE			SZ_128K
+#define ARC2_DCCM_SIZE			SZ_64K
+#define ARC0_MEM_SIZE			(ARC0_ICCM_SIZE + ARC0_DCCM_SIZE)
+#define ARC1_MEM_SIZE			(ARC1_ICCM_SIZE + ARC1_DCCM_SIZE)
+#define ARC2_MEM_SIZE			(ARC2_ICCM_SIZE + ARC2_DCCM_SIZE)
+#define HL_FW_SRAM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_FWSRAM_OFFSET)
+#define HL_FW_SRAM_SIZE			SZ_128K
+
+#endif /* _SOC_H_ */
